src/Jobs/UpdateAccessTokensLastUsed.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Butler\Auth\Jobs;
+
+use Butler\Auth\AccessToken;
+use Butler\Auth\Facades\TokenCache;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+
+class UpdateAccessTokensLastUsed implements ShouldQueue, ShouldBeUnique
+{
+    use InteractsWithQueue;
+    use Dispatchable;
+    use Queueable;
+    use SerializesModels;
+
+    public function handle()
+    {
+        foreach (AccessToken::all() as $token) {
+            if ($cachedToken = TokenCache::get($token->token)) {
```

Comment:
```suggestion
            if ($cachedToken = TokenCache::get($token->token) && $cachedToken->last_used_at < $cachedToken->last_used_at) {
```

Since this might be run as a job we shouldn't update the `last_used_at` field if it's _after_ the cached value (i.e. another job has already updated with a date after the one available in this job)

Reply:
Updated.